internal/impl/qdrant/value.go

```diff
@@ -0,0 +1,178 @@
+package qdrant
+
+// This file contains methods to convert a generic map to map[string]*pb.Value.
+// Qdrant uses this format for its JSON payload.
+// Proto definition: https://github.com/qdrant/qdrant/blob/master/lib/api/src/grpc/proto/json_with_int.proto
+// This is a custom implementatation based on "google.golang.org/protobuf/types/known/structpb".
+// It extends the original implementation to support IntegerValue and DoubleValue as Qdrant requires.
+//
+// USAGE:
+//
+// jsonMap := map[string]interface{}{
+// 	"some_null":    nil,
+// 	"some_bool":    true,
+// 	"some_int":     42,
+// 	"some_float":   3.14,
+// 	"some_string":  "hello",
+// 	"some_bytes":   []byte("world"),
+// 	"some_nested":  map[string]interface{}{"key": "value"},
+// 	"some_list":    []interface{}{"foo", 32},
+// }
+//
+// valueMap := NewValueMap(jsonMap)
+
+import (
+	"encoding/base64"
+	"fmt"
+	"unicode/utf8"
+
+	pb "github.com/qdrant/go-client/qdrant"
+)
+
+// Converts a map of string to interface{} to a map of string to *grpc.Value
+//
+//	╔════════════════════════╤════════════════════════════════════════════╗
+//	║ Go type                │ Conversion                                 ║
+//	╠════════════════════════╪════════════════════════════════════════════╣
+//	║ nil                    │ stored as NullValue                        ║
+//	║ bool                   │ stored as BoolValue                        ║
+//	║ int, int32, int64      │ stored as IntegerValue                     ║
+//	║ uint, uint32, uint64   │ stored as IntegerValue                     ║
+//	║ float32, float64       │ stored as DoubleValue                      ║
+//	║ string                 │ stored as StringValue; must be valid UTF-8 ║
+//	║ []byte                 │ stored as StringValue; base64-encoded      ║
+//	║ map[string]interface{} │ stored as StructValue                      ║
+//	║ []interface{}          │ stored as ListValue                        ║
+//	╚════════════════════════╧════════════════════════════════════════════╝
+
+func newValueMap(input any) (map[string]*pb.Value, error) {
+	valueMap := make(map[string]*pb.Value)
```

Comment:
Small nit: `make(map[string]*pb.Value, len(inputMap))` and place it after the type assertion.